src/main/java/com/notably/model/suggestion/SuggestionModel.java

```diff
@@ -0,0 +1,49 @@
+package com.notably.model.suggestion;
+
+import java.util.List;
+import java.util.Optional;
+
+import javafx.beans.property.Property;
+import javafx.collections.ObservableList;
+
+/**
+ * Represents the Suggestion Model instance.
+ * The UI will directly interact with this model to display the suggestions result.
+ */
+interface SuggestionModel {
+    /**
+     * Gets the list of suggestions saved in the model.
+     * @return The Observable List of suggested items.
+     */
+    ObservableList<SuggestionItem> getSuggestions();
+
+    /**
+     * Saves the list of suggestions in the model.
+     * @param suggestions The list of paths or notes that match the user input.
+     */
+    void setSuggestions(List<SuggestionItem> suggestions);
+
+    /**
+     * Shows user the meaning of their input, e.g. "Create a new note".
+     * When input cannot get corrected, the "error" message will be shown in commandTextProperty.
+     * TODO: add logo.
+     * @return The meaning of the user input.
+     */
+    Property<Optional<String>> commandTextProperty();
```

Comment:
I would like to clarify, the "error" message is just a specific String? Or is it the case that if there exists no correction, then an Optional empty is returned? 

Also, I think it is better to avoid Optional<String> in the context of String Property. I'm actually not sure if Optionals are supported as Property types, could someone clarify this? How about returning a Property<String> or SimpleStringProperty?

Reply:
The "error" message refers to something like "Do you mean...?". Optional empty is returned when the user deletes all of his input in the text box.

As of the `Property` and `Optional<String>` issue, I'm not entirely sure about it. Maybe @kevinputera can help?

Reply:
@HemanshuGandhi Yep! The error message here is just a specific string. We can extend more on this in the future if we would like to display more complex feedbacks!

Optional empty is returned when there isn't any feedback to be shown to the user; that is when the TextField is empty.

The `Optional<String>` type here was actually chosen as it can represent the "empty" state really well. We can certainly go with `Property<String>` instead, but then the "empty" state won't be conveyed as well (we can certainly go with an empty string "", but isn't `Optional.empty()` more descriptive?).

As for whether or not `Property<Optional<String>>` is supported, I believe so! From my understanding, the `Property` class simply encapsulates the business logic required to achieve the observability behavior; it doesn't dictate which classes are supported!

Reply:
alright, thanks for clarifying. Lets stick to the current one then.